pkg/volume/volume.go

```diff
@@ -145,28 +144,28 @@ type Attacher interface {
 	// GetDeviceMountPath returns a path where the device should
 	// be mounted after it is attached. This is a global mount
 	// point which should be bind mounted for individual volumes.
-	GetDeviceMountPath(spec *Spec) string
+	GetDeviceMountPath(host VolumeHost, spec *Spec) string
 
 	// MountDevice mounts the disk to a global path which
 	// individual pods can then bind mount
-	MountDevice(devicePath string, deviceMountPath string, mounter mount.Interface) error
+	MountDevice(spec *Spec, devicePath string, deviceMountPath string, mounter mount.Interface) error
 }
 
 // Detacher can detach a volume from a node.
 type Detacher interface {
 
 	// Detach the given volume from the given host.
-	Detach(deviceMountPath string, hostName string) error
+	Detach(cloudProvider cloudprovider.Interface, deviceMountPath string, hostName string) error
```

Comment:
Not all volume plugins will have a cloud provider. So we should not require one in the Detach interface. Plugins that require one can get it through the VolumeHost.